src/content/services/monetizationLinkManager.ts

```diff
@@ -40,7 +39,7 @@ export class MonetizationLinkManager extends EventEmitter {
     });
 
     this.documentObserver = new MutationObserver((records) =>
-      this.onWholeDocumentObserved(records),
+      this.onAddedOrRemovedNodesMutationObserverCallback(records),
```

Comment:
I think can keep calling it `onWholeDocumentObserved` still. Just its inner working has changed.